cmd/secret-collection-manager/main_test.go

```diff
@@ -0,0 +1,371 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/vault/api"
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/openshift/ci-tools/pkg/vaultclient"
+)
+
+const (
+	vaultTestingToken = "jpuxZFWWFW7vM882GGX2aWOE"
+	vaultAddr         = "127.0.0.1:8302"
+	managerListenAddr = "127.0.0.1:8080"
+)
+
+func TestSecretCollectionManager(t *testing.T) {
+	if _, err := exec.LookPath("vault"); err != nil {
+		if _, runningInCi := os.LookupEnv("CI"); runningInCi {
+			t.Fatalf("could not find vault in path: %v", err)
+		}
+		t.Skip("could not find vault in path")
+	}
+	if os.Getenv("CI") != "" {
+		// We need a writeable home
+		os.Setenv("HOME", "/tmp")
+	}
+	logrus.SetLevel(logrus.TraceLevel)
+
+	vaultCancel, vaultDone, err := startVault(t)
+	if err != nil {
+		t.Fatalf("failed to start vault: %v", err)
+	}
+	t.Cleanup(func() {
+		vaultCancel()
+		<-vaultDone
+	})
+
+	client, err := vaultclient.New("http://"+vaultAddr, vaultTestingToken)
+	if err != nil {
+		t.Fatalf("failed to construct vault client: %v", err)
+	}
+
+	if err := client.Sys().EnableAuthWithOptions("userpass", &api.EnableAuthOptions{Type: "userpass"}); err != nil {
+		t.Fatalf("failed to enable userpass auth: %v", err)
+	}
+
+	mounts, err := client.ListAuthMounts()
+	if err != nil {
+		t.Fatalf("failed to list auth mounts: %v", err)
+	}
+	var mountAccessor string
+	for _, mount := range mounts {
+		if mount.Type == "userpass" {
+			mountAccessor = mount.Accessor
+			break
+		}
+	}
+	if mountAccessor == "" {
+		t.Fatalf("failed to find userpass mount")
+	}
+
+	for _, user := range []string{"user-1", "user-2"} {
+		if _, err := client.Logical().Write(fmt.Sprintf("/auth/userpass/users/%s", user), map[string]interface{}{"password": "password"}); err != nil {
+			t.Fatalf("failed to create userpass user %s: %v", user, err)
+		}
+		identity, err := client.CreateIdentity(user, []string{"default"})
+		if err != nil {
+			t.Fatalf("failed to create identity for user %s: %v", user, err)
+		}
+		if _, err := client.Logical().Write("identity/entity-alias", map[string]interface{}{
+			"name":           user,
+			"canonical_id":   identity.ID,
+			"mount_accessor": mountAccessor,
+		}); err != nil {
+			t.Fatalf("failed to create identity alias for user %s in mount_accessor %s: %v", user, mountAccessor, err)
+		}
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	server := server(client, "secret/self-managed", managerListenAddr)
+	go func() {
+		if err := server.ListenAndServe(); err != nil && ctx.Err() == nil {
+			cancel()
+			t.Errorf("failed to start secret-collection-manager: %v", err)
+		}
+	}()
+
+	if err := waitForServiceReady(ctx, t, fmt.Sprintf("http://%s/healthz", managerListenAddr)); err != nil {
```

Comment:
The e2e framework has code to transparently stop, wait and stop (on timeout) processes that use the k8s ready/health endpoints (so it knows which ports to poke) - might be able to re-use?